File: phone-append-2-nodejs/REST/pa2_response.js
/**
 * Response from PhoneAppend API, containing phone information and potential error.
 */
export class PA2Response {
    constructor(data = {}) {
        this.PhoneInfo = data.PhoneInfo ? new PhoneInfo(data.PhoneInfo) : null;
        this.Error = data.Error ? new Error(data.Error) : null;
    }

    toString() {
        return `PA2Response: PhoneInfo = ${this.PhoneInfo ? this.PhoneInfo.toString() : 'null'}, Error = ${this.Error ? this.Error.toString() : 'null'}`;
    }
}

/**
 * Phone information for API responses.
 */
export class PhoneInfo {
    constructor(data = {}) {
        this.Phone = data.Phone;
        this.Name = data.Name;
        this.Address = data.Address;
        this.City = data.City;
        this.State = data.State;
        this.PostalCode = data.PostalCode;
        this.IsResidential = data.IsResidential;
        this.Certainty = data.Certainty;
        this.LineType = data.LineType;
    }

    toString() {
        return `PhoneInfo: Phone = ${this.Phone}, Name = ${this.Name}, Address = ${this.Address}, City = ${this.City}, State = ${this.State}, PostalCode = ${this.PostalCode}, IsResidential = ${this.IsResidential}, Certainty = ${this.Certainty}, LineType = ${this.LineType}`;
    }
}

/**
 * Error object for API responses.
 */
export class Error {
    constructor(data = {}) {
        this.Type = data.Type;
        this.TypeCode = data.TypeCode;
        this.Desc = data.Desc;
        this.DescCode = data.DescCode;
    }

    toString() {
        return `Error: Type = ${this.Type}, TypeCode = ${this.TypeCode}, Desc = ${this.Desc}, DescCode = ${this.DescCode}`;
    }
}

export default PA2Response;

File: phone-append-2-python/REST/pa2_response.py
from dataclasses import dataclass
from typing import Optional, List

# Input parameters for the PhoneAppend API call.
@dataclass
class PhoneAppendInput:
    FullName: Optional[str] = None
    FirstName: Optional[str] = None
    LastName: Optional[str] = None
    Address: Optional[str] = None
    City: Optional[str] = None
    State: Optional[str] = None
    PostalCode: Optional[str] = None
    IsBusiness: Optional[str] = None
    LicenseKey: Optional[str] = None

    def __str__(self) -> str:
        return (f"PhoneAppendInput: FullName={self.FullName}, FirstName={self.FirstName}, "
                f"LastName={self.LastName}, Address={self.Address}, City={self.City}, "
                f"State={self.State}, PostalCode={self.PostalCode}, "
                f"IsBusiness={self.IsBusiness}, LicenseKey={self.LicenseKey}")

# Error object for API responses.
@dataclass
class Error:
    Type: Optional[str] = None
    TypeCode: Optional[str] = None
    Desc: Optional[str] = None
    DescCode: Optional[str] = None

    def __str__(self) -> str:
        return (f"Error: Type={self.Type}, TypeCode={self.TypeCode}, "
                f"Desc={self.Desc}, DescCode={self.DescCode}")

# Phone information for API responses.
@dataclass
class PhoneInfo:
    Phone: Optional[str] = None
    Name: Optional[str] = None
    Address: Optional[str] = None
    City: Optional[str] = None
    State: Optional[str] = None
    PostalCode: Optional[str] = None
    IsResidential: Optional[str] = None
    Certainty: Optional[str] = None
    LineType: Optional[str] = None
    Debug: Optional[str] = None

    def __str__(self) -> str:
        return (f"PhoneInfo: Phone={self.Phone}, Name={self.Name}, "
                f"Address={self.Address}, City={self.City}, State={self.State}, "
                f"PostalCode={self.PostalCode}, IsResidential={self.IsResidential}, "
                f"Certainty={self.Certainty}, LineType={self.LineType}, Debug={self.Debug}")

# Response from PhoneAppend API, containing phone information and potential error.
@dataclass
class PA2Response:
    PhoneInfo: Optional['PhoneInfo'] = None
    Error: Optional['Error'] = None

    def __str__(self) -> str:
        phone_info = str(self.PhoneInfo) if self.PhoneInfo else 'None'
        error = str(self.Error) if self.Error else 'None'
        return f"PA2Response: PhoneInfo={phone_info}, Error={error}"